File: BulletPlugin/src/BulletShapeFileManager.cpp
#include "StdAfx.h"
#include "..\include\BulletShapeFileManager.h"
#include "BulletShapeBuilder.h"
#include "BulletShapeRepository.h"

namespace BulletPlugin
{

BulletShapeFileManager::BulletShapeFileManager(void)
:shapeRepository(gcnew BulletShapeRepository()),
shapeBuilder(gcnew BulletShapeBuilder()),
ShapeFileManager(shapeRepository, shapeBuilder)
{
	shapeBuilder->setRepository(shapeRepository);
}

void BulletShapeFileManager::loadStarted()
{

}

void BulletShapeFileManager::loadEnded()
{

}

float BulletShapeFileManager::ShapeMargin::get()
{
	return shapeBuilder->ShapeMargin;
}
void BulletShapeFileManager::ShapeMargin::set(float value)
{
	shapeBuilder->ShapeMargin = value;
}

}

File: BulletWrapperPlugin/ShapeLoading/BulletShapeFileManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Engine;

namespace BulletPlugin
{
    class BulletShapeFileManager : ShapeFileManager
    {
        public BulletShapeFileManager(BulletShapeRepository repository, BulletShapeBuilder builder)
            :base(repository, builder)
        {
            ShapeBuilder = builder;
            ShapeRepository = repository;
            ShapeBuilder.setRepository(ShapeRepository);
        }

        protected override void loadStarted()
        {
            
        }

        protected override void loadEnded()
        {
            
        }

        public BulletShapeRepository ShapeRepository { get; private set; }

        public BulletShapeBuilder ShapeBuilder { get; private set; }
    }
}
